src/solver/utils/ortools_utils.cpp

```diff
@@ -38,6 +42,56 @@ static void setGenericParameters(MPSolverParameters& params)
     params.SetIntegerParam(MPSolverParameters::PRESOLVE, 0);
 }
 
+static void checkSetSolverSpecificParameters(bool status,
+                                             const std::string& solverName,
+                                             const std::string& specificParameters)
+{
+    if (!status)
+    {
+        throw Antares::Error::InvalidSolverSpecificParameters(solverName, specificParameters);
+    }
+    else
+    {
+        Antares::logs.info() << "  Successfully set " + solverName + " solver specific parameters";
+    }
+}
+
+static void TuneSolverSpecificOptions(
+  MPSolver* solver,
+  const std::string& solverName,
+  const Antares::Solver::Optimization::SolverParameters& solverParameters)
+{
+    if (!solver)
+    {
+        return;
+    }
+
+    bool status;
+    std::string specificParams;
+
+    switch (solver->ProblemType())
+    {
+    // Allow solver to use only one thread
+    case MPSolver::XPRESS_LINEAR_PROGRAMMING:
+    case MPSolver::XPRESS_MIXED_INTEGER_PROGRAMMING:
+    {
+        specificParams = XPRESS_PARAMS + " " + solverParameters.xpress;
+        status = solver->SetSolverSpecificParametersAsString(specificParams);
+        checkSetSolverSpecificParameters(status, solverName, specificParams);
+        break;
+    }
+    case MPSolver::SCIP_MIXED_INTEGER_PROGRAMMING:
+    {
+        specificParams = SCIP_PARAMS + ", " + solverParameters.scip;
```

Comment:
I think SCIP_PARAMS/XPRESS_PARAMS should be overriden (ie ignored) if the user provides custom parameters.

Otherwise it seems impossible to set a different value for the number of threads

Reply:
I think it works as is as the `SetSolverSpecificParameters` will set parameters in parsing order so in theory threads can be set twice (one from XPRESS/SCIP PARAMS, and one from user specific choice). But this is really unclear, so I will change according to your suggestion

Reply:
Finally, following our discussion, I will keep the current version of the code as the behavior is ok